### PR TITLE
[backport] mopidy: fix, resolves #32234

### LINKED ISSUE #32234: Mopidy is broken
## Issue description
After upgrading my system to `nixpkgs` commit cc1d7a358f5, `mopidy` fails to start
```
● mopidy.service - mopidy music player daemon
   Loaded: loaded (/nix/store/4jwmkklrip1nkkxsi558cl124f8f0miv-unit-mopidy.service/mopidy.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Fri 2017-12-01 16:54:41 CET; 19s ago
  Process: 8885 ExecStart=/nix/store/xxgp8r8m8af3mfzqc0bi0zgvcnc1cm5c-python-2.7.14-env/bin/mopidy --config /nix/store/<hidden>-mopidy.conf (code=exited, status=203/EXEC)
  Process: 8877 ExecStartPre=/nix/store/8zcsgrsj4hj7jgjjlwdpiwhk6hcv7vm5-unit-script/bin/mopidy-pre-start (code=exited, status=0/SUCCESS)
 Main PID: 8885 (code=exited, status=203/EXEC)
      CPU: 21ms

Dec 01 16:54:41 neon systemd[1]: Starting mopidy music player daemon...
Dec 01 16:54:41 neon systemd[1]: Started mopidy music player daemon.
Dec 01 16:54:41 neon systemd[8885]: mopidy.service: Failed at step EXEC spawning /nix/store/xxgp8r8m8af3mfzqc0bi0zgvcnc1cm5c-python-2.7.14-env/bin/mopidy: No such file or directory
```

### Steps to reproduce
checkout cc1d7a358f5 , set `services.mopidy.enable = true`

## Technical details
```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.9.65, NixOS, 18.03pre121732.cc1d7a358f5 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.15`
 - channels(root): `"nixos-18.03pre121732.cc1d7a358f5, nixpkgs-18.03pre114739.d0d905668c"`
 - channels(rvolosatovs): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`
```

### LINKED ISSUE #36205: services.mopidy.enable = true; fails
## Issue description

When I try to enable mopidy, I get **exactly** the same error as https://github.com/NixOS/nixpkgs/issues/33932:
```
collision between `/nix/store/3d8nq34016rjz5vfy7w36k77nnrvkchk-gst-python-1.10.4/lib/python2.7/site-packages/gi/overrides/__init__.pyo' and `/nix/store/xir2hwxj8870j311hx8jdi63vcspljs0-python2.7-pygobject-3.24.1/lib/python2.7/site-packages/gi/overrides/__init__.pyo'
```

Is this a known issue?

Full error message:
```
$ sudo nixos-rebuild boot
building Nix...
warning: in configuration file '/etc/nix/nix.conf': unknown setting 'signed-binary-caches'
warning: in configuration file '/etc/nix/nix.conf': unknown setting 'signed-binary-caches'
building the system configuration...
warning: in configuration file '/etc/nix/nix.conf': unknown setting 'signed-binary-caches'
these derivations will be built:
  /nix/store/87vchvqv4misz0qxhh7fmxbwp3dihln0-system-path.drv
  /nix/store/08wzmf5v9387449kks199ar2x1j6jl3g-unit-polkit.service.drv
  /nix/store/2h60dd3whbfrsqv1vaklsn71mg93s4lj-dbus-1.drv
  /nix/store/ddsfxdccab6mv1bvxk25k8pf3pp6i2k9-unit-dbus.service.drv
  /nix/store/9ac0dynaq8cm8n8280wmph204fl87hff-user-units.drv
  /nix/store/flrd0spl6qqkssbvfknlrq40g5gccsjw-unit-systemd-fsck-.service.drv
  /nix/store/z32xpl22v64vhw91pcd3d7iyjl2hp1vb-python-2.7.14-env.drv
  /nix/store/lxmzpva4spmz3l4ny79r3lqcwirjsk1x-unit-mopidy-scan.service.drv
  /nix/store/z583bmd414jflbd33d440zgpkzpg44d2-unit-mopidy.service.drv
  /nix/store/hbyfiv3x1hq4gm7jdcsd77fhjwjmzjjf-system-units.drv
  /nix/store/p3jf9jjf8ws884cs20jk2h6ki2yg6zks-etc.drv
  /nix/store/2x07d2aqs981hvkd148wmssylw6wpqxm-nixos-system-dn-17.09.3109.48a76ca6f57.drv
building '/nix/store/87vchvqv4misz0qxhh7fmxbwp3dihln0-system-path.drv'...
building '/nix/store/z32xpl22v64vhw91pcd3d7iyjl2hp1vb-python-2.7.14-env.drv'...
collision between `/nix/store/3d8nq34016rjz5vfy7w36k77nnrvkchk-gst-python-1.10.4/lib/python2.7/site-packages/gi/overrides/__init__.pyo' and `/nix/store/xir2hwxj8870j311hx8jdi63vcspljs0-python2.7-pygobject-3.24.1/lib/python2.7/site-packages/gi/overrides/__init__.pyo'
builder for '/nix/store/z32xpl22v64vhw91pcd3d7iyjl2hp1vb-python-2.7.14-env.drv' failed with exit code 25
cannot build derivation '/nix/store/lxmzpva4spmz3l4ny79r3lqcwirjsk1x-unit-mopidy-scan.service.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/z583bmd414jflbd33d440zgpkzpg44d2-unit-mopidy.service.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/hbyfiv3x1hq4gm7jdcsd77fhjwjmzjjf-system-units.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/p3jf9jjf8ws884cs20jk2h6ki2yg6zks-etc.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/2x07d2aqs981hvkd148wmssylw6wpqxm-nixos-system-dn-17.09.3109.48a76ca6f57.drv': 1 dependencies couldn't be built
error: build of '/nix/store/2x07d2aqs981hvkd148wmssylw6wpqxm-nixos-system-dn-17.09.3109.48a76ca6f57.drv' failed
```

### Steps to reproduce

AFAIK this triggers it:
```
{
  services.mopidy.enable = true;
}
```
But I can't rule out the possibility that something else is interfering with mopidy's build (is that even possible?).

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

 - system: `"x86_64-linux"`
 - host os: `Linux 4.9.82, NixOS, 17.09.3047.8bce347f02f (Hummingbird)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 1.12pre5873_b76e282d`
 - channels(root): `"nixos-17.09.3109.48a76ca6f57"`
 - channels(delta): `"nixpkgs-18.03pre129212.098c7f3d945"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change
Fix #36205 

###### Things done
- [x] Tested mopidy.service runs in NixOS VM
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/audio/mopidy.nix [hunk 1]
```diff
@@ -4,17 +4,22 @@ with pkgs;
...
+    name = "mopidy-with-extensions-${mopidy.version}";
+    paths = closePropagation cfg.extensionPackages;
+    pathsToLink = [ "/${python.sitePackages}" ];
+    buildInputs = [ makeWrapper ];
+    postBuild = ''
+      makeWrapper ${mopidy}/bin/mopidy $out/bin/mopidy \
+        --prefix PYTHONPATH : $out/${python.sitePackages}
+    '';
   };
-
```
Comment (Mic92) [inline nixos/modules/services/audio/mopidy.nix]: What is the commit hash from master for this backport?

Comment (Mic92) [reply nixos/modules/services/audio/mopidy.nix]: 85b84527f636c60bd8c0f0567bb471d491fb5a89

**Comment by Mic92** [comment]
pushed in `2070830ba845b69c7c597537786d9b9498fa4b28`
For backports we use `git cherry-pick -x` to include the original commit hash in the message.

**Comment by rnhmjoj** [comment]
Ok, thank you.

**Pull request closed without merging**